sqlglot/parser.py

```diff
@@ -1234,6 +1237,20 @@ def _parse_create(self) -> t.Optional[exp.Expression]:
                 if self._match_text_seq("WITH", "NO", "SCHEMA", "BINDING"):
                     no_schema_binding = True
 
+            if self._match_text_seq("CLONE"):
+                clone = self._parse_table(schema=True)
+                when = self._match_texts({"AT", "BEFORE"}) and self._prev.text
+                clone_kind = (
+                    self._match(TokenType.L_PAREN)
+                    and self._match_texts(self.CLONE_KINDS)
+                    and self._prev.text
```

Comment:
we should normalize to upper(), not sure if we should do in parser or gen

Reply:
sounds good, I think either works fine because it's probably a snowflake-specific thing